### PR TITLE
Upgrade to pylint 2.12 and enable ``for_any_all`` checker

### PR DESCRIPTION
**Pull request by nickdrozd**
|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
This is great, thank you ! Could you also fix the other issues newly detected by pylint 2.12 ? (Or disable them if you don't have the time right now)

**Comment by nickdrozd** [comment]
Looks like the new version of Pylint triggers a few issues.

File: astroid/mixins.py [hunk 1]
```diff
@@ -28,6 +28,8 @@
 class BlockRangeMixIn:
```
Comment (Pierre-Sassoulas) [inline astroid/mixins.py:28]: This mixin is not recognized as one, I'll fix this myself suggestions won't work here.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

File: astroid/nodes/node_ng.py [hunk 2]
```diff
@@ -448,7 +445,7 @@ def _fixed_source_line(self) -> Optional[int]:
         We need this method since not all nodes have :attr:`lineno` set.
         """
         line = self.lineno
-        _node: Optional[NodeNG] = self
+        _node: Optional[NodeNG] = self  # pylint: disable = used-before-assignment
```
Comment (DanielNoord) [inline astroid/nodes/node_ng.py:448]: Is this a new false positive?

Comment (cdce8p) [reply astroid/nodes/node_ng.py:448]: https://github.com/PyCQA/pylint/issues/5342

File: pylintrc [hunk 3]
```diff
@@ -301,6 +301,9 @@ ignored-modules=typed_ast.ast3
 # (useful for classes with attributes dynamically set).
 ignored-classes=SQLObject
 
+# Regex pattern to define which classes are considered mixins.
+mixin-class-rgx=.*Mix[i|I]n
```
Comment (cdce8p) [inline pylintrc:305]: @Pierre-Sassoulas This will also match `Mix|n`. It should just be `.*Mix[Ii]n`.

File: astroid/nodes/node_ng.py (hunk 2 above)
Comment (DanielNoord) [reply astroid/nodes/node_ng.py:448]: Ah! Thanks!

File: pylintrc (hunk 3 above)
Comment (Pierre-Sassoulas) [reply pylintrc:305]: 😄 Obviously I'm just handing the case with really big using pipe, and did not make a regex mistake at all.

**Pull request merged**